### DISCUSSION TITLE
Increase bitmap quality (enable anti aliasing)

### DISCUSSION THREAD

**Discussion by JanMann89** [stephanrauh/ngx-extended-pdf-viewer#1758] [Q&A]
Hi!

The PDFs, that I need to display in my project, usually contain bunches of text as bitmaps. By default the ngx-extended-pdf-viewer displays the text, especially in small zoom levels, very pixelated. So I wonder wether it is possible to activate anti aliasing.

I already tried to activate the svg renderer by using `pdfDefaultOptions.renderer = 'svg'`, but there are two problems:
1. SVG renderer is marked as deprecated
2. problems rending PDFs with integrated fonts

I would really appreciate if someone can help.

//Additions:
- The pixelated rendering seems to be in Chrome and Safari, not in Firefox.
- I use ngx-extended-pdf-viewer 16.2.16 in Angular 14.
- Example that shows the rendering difference:
![image](https://user-images.githubusercontent.com/51460788/237057361-ca93ade0-6ed5-4dc0-b8a0-78d581d9598f.png)

Thanks in advance,
Jan

**Comment by stephanrauh** [comment]
Can you check if the quality is bad on the Mozilla showcase, too? I suspect it's a problem of the core library. https://mozilla.github.io/pdf.js/web/viewer.html

**Reply by JanMann89** [reply]
Since the Problem only affects bitmaps inside the pdf and I can only find vector images inside the Mozilla showcase, there is no lack of quality in that document.

//edit:
Ah, sorry, I got you wrong in the first place. I loaded my testfile into the viewer you posted and the quality issue is still there.

**Reply by stephanrauh** [reply]
@timvandermeij Jan reports that embedded images don't scale as nicely in pdf.js as they do in Acrobat. Does this ring a bell? Is there an option to enable anti-aliasing or any other magic? You'll find the full details further up in this ticket.

**Reply by JanMann89** [reply]
Yeah, I just wonder wether it is possible, because the pdf-js-viewer has antialiasing enabled and it uses the same core library, as far as I know. It uses also the canvas renderer of pdf.js. Unfortunately I can not use that library because of other restrictions, such as needed unsafe CSP policies.

**Reply by stephanrauh** [reply]
You're talking about this library? https://github.com/intbot/ng2-pdfjs-viewer

**Reply by JanMann89** [reply]
> You're talking about this library? https://github.com/intbot/ng2-pdfjs-viewer

Yes, exactly.

**Reply by timvandermeij** [reply]
This doesn't immediately ring a bell for me. Given that you indicate that it doesn't happen in Firefox but it does happen in other browsers I suspect this is either a browser bug or a feature we rely on that only happens to work in Firefox. In any case, I would recommend to open a bug at https://github.com/mozilla/pdf.js/issues with an example PDF file since that is important in order to be able to reproduce the issue.
